### PR TITLE
Add documentation about the QGIS Server access control plugins

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: source/docs/pyqgis_developer_cookbook/server.rst [hunk 1]
```diff
@@ -276,3 +276,133 @@ The watermark filter example shows how to replace the WMS output with a new imag
...
+           return super(QgsAccessControlFilter, self).allowToEdit(layer, feature)
+
+       def cacheKey(self):
+           return super(QgsAccessControlFilter, self).cacheKey()
+
+This example gives a full access for everybody.
+
+It's the role of the plugin to know who is logged on.
+
+On all those method we have the layer on argument to be able to customise
```
Comment (yjacolin) [inline source/docs/pyqgis_developer_cookbook/server.rst]: methods ?

File: source/docs/pyqgis_developer_cookbook/server.rst [hunk 2]
```diff
@@ -276,3 +276,133 @@ The watermark filter example shows how to replace the WMS output with a new imag
...
+   def allowToEdit(self, layer, feature):
+       return feature.attribute('role') == 'user'
+
+To be able to edit only feature that has the attribute role
+with the value user.
+
+cacheKey
+--------
+
+QGIS server maintain a cache of the capabilities than to have a cache
```
Comment (yjacolin) [inline source/docs/pyqgis_developer_cookbook/server.rst]: s/than/then/ ?

**Comment by elpaso** [comment]
Would you please add some introductory text about AC plugins? 
IMHO it should be at the beginning of the plugins section and later on, just before you start talking about AC plugins.
I doubt that if left as it is now anybody will understand what that access control is all about.

**Comment by sbrunner** [comment]
Pull request updated :-)

**Comment by elpaso** [comment]
Sorry to bother you again. I might be wrong but I feel that there is still one missing piece (registering of AC filters from a plugin). I think that in general you should ask yourself if the reader will be able to write a new AC plugin from scratches after reading this page.

**Comment by sbrunner** [comment]
Oups effectively, thanks @elpaso :-)

**Comment by sbrunner** [comment]
Updated :-)

**Comment by sbrunner** [comment]
Thanks @elpaso

**Pull request merged**